lib/include/tree_sitter/parser.h

```diff
@@ -44,6 +44,7 @@ struct TSLexer {
   TSSymbol result_symbol;
   void (*advance)(TSLexer *, bool);
   void (*mark_end)(TSLexer *);
+  uint32_t* line;
```

Comment:
Note: changing this to a getter could be worth the indirection if in the future this field might get a dynamic address or be computed on-demand

Reply:
There are two types of objects in scanner's API, method pointers and values, this change doesn't cost to introduce a third type, a pointer to a value. Also to minimize breaking changes it makes sense to implement the final best solution at once.

Reply:
Yup, I realized the direct pointer was a mistake too, just changed it to a getter